Volume/Views/Orgs Login/OrgsLoginView.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  OrgsLoginView.swift
+//  Volume
+//
+//  Created by Vin Bui on 9/29/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Combine
+import SwiftUI
+
+struct OrgsLoginView: View {
+    
+    // MARK: - Properties
+    
+    @Environment(\.dismiss) var dismiss
+    @StateObject var viewModel = ViewModel()
+    
+    @FocusState private var accessCodeIsFocused: Bool
+    @FocusState private var slugIsFocused: Bool
+    
+    // MARK: - Constants
+    
+    private struct Constants {
+        static let buttonTextFont: Font = .helveticaNeueMedium(size: 16)
+        static let errorMessageFont: Font = .helveticaNeueMedium(size: 14)
+        static let inputSpacing: CGFloat = 24
+        static let labelFont: Font = .newYorkRegular(size: 16)
+        static let labelSpacing: CGFloat = 8
+        static let maxAccessCodeLength: Int = 6
+        static let navHeaderText: Font = .newYorkMedium(size: 20)
+        static let sidePadding: CGFloat = 16
+        static let textColor: Color = Color.volume.textGray
+        static let textFieldBorderColor: Color = Color.volume.outlineGray
+        static let textFieldFont: Font = .helveticaRegular(size: 16)
+        static let textFieldPadding: EdgeInsets = .init(top: 12, leading: 16, bottom: 12, trailing: 16)
+        static let topPadding: CGFloat = 28
+    }
+    
+    // MARK: - UI
+    
+    var body: some View {
+        VStack(spacing: Constants.inputSpacing) {
+            slugInput
+            codeInput
+            
+            VStack(alignment: .leading, spacing: 8) {
+                authenticateButton
+                
+                viewModel.showErrorMessage ? errorMessage : nil
+            }
+            
+            Spacer()
+        }
+        .padding(.top, Constants.topPadding)
+        .padding(.horizontal, Constants.sidePadding)
+        .toolbar {
+            ToolbarItem(placement: .principal) {
+                Text("Organization Login")
+                    .font(Constants.navHeaderText)
+            }
+            
+            ToolbarItem(placement: .navigationBarLeading) {
+                Button {
+                    dismiss()
+                } label: {
+                    Image.volume.backArrow
+                }
+                .buttonStyle(EmptyButtonStyle())
+            }
+        }
+        .navigationBarBackButtonHidden(true)
+        .background(Color.volume.backgroundGray)
+        .onChange(of: viewModel.accessCode) { _ in
+            withAnimation(.easeOut(duration: 0.3)) {
+                viewModel.buttonEnabled = viewModel.accessCode.count == Constants.maxAccessCodeLength && !viewModel.slug.isEmpty
+            }
+        }
+        .onChange(of: viewModel.slug) { _ in
+            withAnimation(.easeOut(duration: 0.3)) {
+                viewModel.buttonEnabled = viewModel.accessCode.count == Constants.maxAccessCodeLength && !viewModel.slug.isEmpty
```

Comment:
Factor out as `updateAuthenticationButton()` in `OrgsLoginModel`

Reply:
In next PR